### PR TITLE
Eth2api: `GetBlockSSZ` supports bellatrix block

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds support for Bellatrix block to `GetBlockSSZ` eth2api implementation. 
The addition is from the `kintsugi` branch. The original author is @potuz, and I added the tests

### CODE REVIEW

File: beacon-chain/rpc/apimiddleware/custom_hooks.go [hunk 1]
```diff
@@ -312,6 +317,14 @@ func serializeV2Block(response interface{}) (apimiddleware.RunDefault, []byte, a
 				Signature: respContainer.Data.Signature,
 			},
 		}
+	} else if strings.EqualFold(respContainer.Version, strings.ToLower(ethpbv2.Version_MERGE.String())) {
```
Comment (potuz) [inline beacon-chain/rpc/apimiddleware/custom_hooks.go:320]: what's the status of `Version_MERGE` vs `Version_Bellatrix`?

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 2]
```diff
@@ -292,21 +292,43 @@ func (bs *Server) GetBlockV2(ctx context.Context, req *ethpbv2.BlockRequestV2) (
...
+	}
+	if bellatrixBlk != nil {
+		v2Blk, err := migration.V1Alpha1BeaconBlockBellatrixToV2(bellatrixBlk.Block)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not get signed beacon block: %v", err)
+		}
+		return &ethpbv2.BlockResponseV2{
+			Version: ethpbv2.Version_MERGE,
+			Data: &ethpbv2.SignedBeaconBlockContainerV2{
+				Message:   &ethpbv2.SignedBeaconBlockContainerV2_MergeBlock{MergeBlock: v2Blk},
```
Comment (potuz) [inline beacon-chain/rpc/eth/beacon/blocks.go:325]: Same here

File: beacon-chain/rpc/eth/beacon/blocks_test.go [hunk 3]
```diff
@@ -108,6 +108,50 @@ func fillDBTestBlocksAltair(ctx context.Context, t *testing.T, beaconDB db.Datab
 	return genBlk, blkContainers
 }
 
+func fillDBTestBlocksBellatrix(ctx context.Context, t *testing.T, beaconDB db.Database) (*ethpbalpha.SignedBeaconBlockMerge, []*ethpbalpha.BeaconBlockContainer) {
```
Comment (potuz) [inline beacon-chain/rpc/eth/beacon/blocks_test.go:111]: Same here

File: beacon-chain/rpc/eth/beacon/blocks_test.go [hunk 4]
```diff
@@ -108,6 +108,50 @@ func fillDBTestBlocksAltair(ctx context.Context, t *testing.T, beaconDB db.Datab
...
+		att1 := util.NewAttestation()
+		att1.Data.Slot = i
+		att1.Data.CommitteeIndex = types.CommitteeIndex(i)
+		att2 := util.NewAttestation()
+		att2.Data.Slot = i
+		att2.Data.CommitteeIndex = types.CommitteeIndex(i + 1)
+		b.Block.Body.Attestations = []*ethpbalpha.Attestation{att1, att2}
+		root, err := b.Block.HashTreeRoot()
+		require.NoError(t, err)
+		signedB, err := wrapper.WrappedMergeSignedBeaconBlock(b)
```
Comment (potuz) [inline beacon-chain/rpc/eth/beacon/blocks_test.go:138]: Same in lines 113, 117, 126, 138

**Review by potuz — COMMENTED** [review]
The PR looks good to me, but it's hard if not impossible to review this without being the reviewer of #10044 and related PRs, I'd suggest @leolara and @rkapka to take a quick look.

**Comment by terencechain** [comment]
> The PR looks good to me, but it's hard if not impossible to review this without being the reviewer of #10044 and related PRs, I'd suggest @leolara and @rkapka to take a quick look.

Copy from Discord:

#10036 is still open. For now, review and approve with this disparity in mind. A simple renaming should not block off the rest of the progress. Regarding this PR, as long as CI/CD is green, we should have complete confidence that it is good to go. I will bump the priority for #10036, if there's no other attempt by the end of next week, I will personally take care of it

**Review by potuz — APPROVED** [review]

**Comment by leolara** [comment]
There will be conflics between this and the stuff I am doing for the name change. In my opinion I would merge this after we change the name for Bellatrix everywhere, untless this is more important to get merge ASAP.

**Comment by leolara** [comment]
There will be conflics between this and the stuff I am doing for the name change. In my opinion I would merge this after we change the name for Bellatrix everywhere, untless this is more important to get merge ASAP.

**Pull request merged**